### PR TITLE
Replacing ES CURL based legos to handle.web_requests

### PR DESCRIPTION
**Pull request by jayasimha-raghavan-unskript**
## Description
Replacing ES CURL Based legos with requests based APIs

### Testing

Screenshot 2022-12-20 at 6 33 34 PM
Screenshot 2022-12-20 at 6 34 22 PM
Screenshot 2022-12-20 at 6 35 57 PM
Screenshot 2022-12-20 at 6 37 41 PM
Screenshot 2022-12-20 at 6 38 30 PM
Screenshot 2022-12-20 at 6 39 14 PM
Screenshot 2022-12-20 at 6 40 04 PM
Screenshot 2022-12-20 at 6 40 45 PM

### Checklist:
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository.

### CODE REVIEW

File: ElasticSearch/legos/elasticsearch_check_health_status/elasticsearch_check_health_status.py [hunk 1]
```diff
@@ -11,58 +11,40 @@
...
-        print("The status for " + d["cluster_name"] + " is " + d["status"])
-        return d
-    except subprocess.CalledProcessError as e:
-        return e.output
+    
+    output = handle.web_request("/_cluster/health?pretty",  # Path
+                                "GET",                      # Method
+                                None)                       # Data
+
+    # es_path = host + ":"+str(port) + "/_cluster/health?pretty"
```
Comment (amit-chandak-unskript) [inline ElasticSearch/legos/elasticsearch_check_health_status/elasticsearch_check_health_status.py:35]: Remove these commented lines

File: ElasticSearch/legos/elasticsearch_check_health_status/elasticsearch_check_health_status.py [hunk 2]
```diff
@@ -11,58 +11,40 @@
...
+    #     result_dict = json.loads(new_result)
+    #     d = {}
+    #     for key in result_dict:
+    #         d[key] = result_dict[key]
+    #     print("The status for " + d["cluster_name"] + " is " + d["status"])
+    #     return d
+    # except subprocess.CalledProcessError as e:
+    #     return e.output
+
+    return output.args
```
Comment (amit-chandak-unskript) [inline ElasticSearch/legos/elasticsearch_check_health_status/elasticsearch_check_health_status.py:50]: why are we returning output.args? output of web_request is the response itself, right?

File: ElasticSearch/legos/elasticsearch_delete_unassigned_shards/elasticsearch_delete_unassigned_shards.py [hunk 3]
```diff
@@ -38,31 +26,37 @@ def elasticsearch_delete_unassigned_shards(handle,
...
+    if len(list_of_shards) != 0:
+        output2 = handle.web_request("/" + list_of_shards, # Path
+                                     "DELETE",  # Method
+                                     None)      # Data
+
+    o = output2.args
+    # es_path = host + ":" + str(port) + "/_cat/shards?v=true&h=index,shard,prirep,state,node,unassigned.reason&s=state&pretty"
+    # es_header = "Authorization: ApiKey" + " " + api_key
+    # first_command = ["curl", "-k", "-XGET", "-H", es_header, es_path]
+    # p1 = subprocess.Popen(first_command, shell=False, stdout=PIPE, stderr=PIPE, universal_newlines=True)
```
Comment (amit-chandak-unskript) [inline ElasticSearch/legos/elasticsearch_delete_unassigned_shards/elasticsearch_delete_unassigned_shards.py:49]: Remove commented code

Comment (jayasimha-raghavan-unskript) [reply ElasticSearch/legos/elasticsearch_delete_unassigned_shards/elasticsearch_delete_unassigned_shards.py]: Done

File: ElasticSearch/legos/elasticsearch_check_health_status/elasticsearch_check_health_status.py (hunk 2 above)
Comment (jayasimha-raghavan-unskript) [reply ElasticSearch/legos/elasticsearch_check_health_status/elasticsearch_check_health_status.py:35]: Screenshot 2022-12-20 at 11 40 57 AM

the Output is of type `subprocess`  by using .args, we capture both stdout and stderr and return to the calling function.

Comment (amit-chandak-unskript) [reply ElasticSearch/legos/elasticsearch_check_health_status/elasticsearch_check_health_status.py:35]: this is not right, so unresolving it.

Comment (jayasimha-raghavan-unskript) [reply ElasticSearch/legos/elasticsearch_check_health_status/elasticsearch_check_health_status.py]: Resovled it. As i am directly using lambda not via a subprocess.

File: ElasticSearch/legos/elasticsearch_delete_unassigned_shards/elasticsearch_delete_unassigned_shards.py [hunk 4]
```diff
@@ -29,40 +17,30 @@ def elasticsearch_delete_unassigned_shards_printer(output):
...
-    p3 = subprocess.Popen(third_command, stdin=p2.stdout, stdout=PIPE, shell=False, universal_newlines=True)
-    var = p2.stdout.read()
-    fourth_command = ["xargs","-i", "curl" ,"-s", "-X", "DELETE",str(url)]
-    subprocess.Popen(fourth_command, stdin=p3.stdout, stdout=PIPE, shell=False, universal_newlines=True)
-    if var == '':
+    output = handle.web_request("/_cat/shards?v=true&h=index,shard,prirep,state,node,unassigned.reason&s=state&pretty",  # Path
+                            "GET",                      # Method
+                            None)                       # Data
+    list_of_shards = []
+    for line in str(output.args).split('\n'):
```
Comment (amit-chandak-unskript) [inline ElasticSearch/legos/elasticsearch_delete_unassigned_shards/elasticsearch_delete_unassigned_shards.py:33]: This is still using args

File: ElasticSearch/legos/elasticsearch_delete_unassigned_shards/elasticsearch_delete_unassigned_shards.py [hunk 5]
```diff
@@ -29,40 +17,30 @@ def elasticsearch_delete_unassigned_shards_printer(output):
...
+        if "UNASSIGNED" in line:
+            list_of_shards.append(line.split(" ")[0])
+
+    if len(list_of_shards) != 0:
+        output2 = handle.web_request("/" + list_of_shards, # Path
+                                     "DELETE",  # Method
+                                     None)      # Data
+
+    o = output2.args
```
Comment (amit-chandak-unskript) [inline ElasticSearch/legos/elasticsearch_delete_unassigned_shards/elasticsearch_delete_unassigned_shards.py:42]: this is still using args

Comment (jayasimha-raghavan-unskript) [reply ElasticSearch/legos/elasticsearch_delete_unassigned_shards/elasticsearch_delete_unassigned_shards.py]: removed

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**